contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2796,6 +2796,10 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 													(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 													 errmsg("Windows login is not supported in babelfish")));
 									}
+									/* If login is a member of sysadmin, creating user for that login should not be allowed. */
+									if (has_privs_of_role(get_role_oid(login->rolename, false), get_sysadmin_oid()))
+										ereport(ERROR, (errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+														errmsg("Cannot create user for sysadmin role.")));
```

Comment:
1. Acc. to TSQL behaviour, Ideally it should not be blocked. Have we considered not blocking it?
2. We have to define/change(if needed) the behaviour after upgrade for the logins for who user is already created and they are member of sysadmin.

Reply:
If you check https://jira.rds.a2z.com/browse/BABEL-4430, we should also block creating a user when its login is already sysadmin.
 
 The reason why this works in TSQL is that it supports having two users as dbo/db_owner.
Since we do not currently support it in Babelfish, we should block creating a user for a login in any database since it's already a member of the sysadmin role.

Reply:
We have blocked granting sysadmin to a login if it already has a user in previous version https://jira.rds.a2z.com/browse/BABEL-3637